src/python/tests/test_genbank_download.py

```diff
@@ -0,0 +1,41 @@
+"""Unit testing of 'ensembl.io.genomio.genbank.download' module.
+
+The unit testing is divided into one test class per submodule/class found in this module, and one test method
+per public function/class method.
+
+Typical usage example::
+    $ pytest test_genbank_download.py
+
+"""
+import filecmp
+from pathlib import Path
+import pytest
+import requests
+
+from ensembl.io.genomio.genbank.download import download_genbank
+
+class TestGenbank_download:
+    """Tests for the 'download_genbank' class"""   
+        
+    data_dir: Path
+    tmp_dir: Path
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self, tmp_dir: Path, files_dir: Path):
+        """Loads necessary fixtures and values as class attributes."""
+        type(self).tmp_dir = tmp_dir
+        type(self).data_dir = files_dir /"genbank_download"
+
+    def test_successful_download(self):
+        """Tests the 'download_genbank()' method.
+    
+        Args:
+        accession: Genbank accession to be downloaded
+        output: path to the expected output file 
+        """
+
+        accession = "CM023531.1"
+        output_file = self.tmp_dir/f"{accession}.gb"
+        download_genbank(accession, output_file)
+        expected_output = self.data_dir/f"{accession}.gb"
+        assert filecmp.cmp(output_file, expected_output)
```

Comment:
Whilst this implementation is completely correct and works, we have recently moved to a slightly different approach to ease the testing code:
```suggestion
@pytest.mark.parametrize(
    "accession",
    [
        ("CM023531.1"),
    ],
)
def test_successful_download(tmp_dir: Path, data_dir: Path, accession: str):
    """Tests the 'download_genbank()' method.

    Args:
        tmp_dir: Session-scoped temporary directory fixture.
        data_dir: Module's test data directory fixture.
        accession: Genbank accession to be downloaded.

    """
    output_file = self.tmp_dir / f"{accession}.gb"
    download_genbank(accession, output_file)
    expected_output = self.data_dir / f"{accession}.gb"
    assert filecmp.cmp(output_file, expected_output)
```
I have opted to add parametrization even in this case to make it easier to add more test cases in the future.

Reply:
Thank you Jorge, I have missed this and it makes sense.